### PR TITLE
feat: Support urls to wasms

### PR DESCRIPTION
**Pull request by bhelx**
Support passing url to wasm in as argument:

```
extism call https://raw.githubusercontent.com/extism/extism/main/wasm/code.wasm count_vowels --input="hello world"
{"count": 3}⏎ 
```

### CODE REVIEW

File: extism_cli/__init__.py [hunk 1]
```diff
@@ -586,11 +586,23 @@ def main():
...
+                from urllib.parse import urlparse
+                pieces = urlparse(args.wasm)
+                if pieces.scheme.lower() in ('http', 'https'):
+                    import urllib.request
+                    f = urllib.request.urlopen(args.wasm)
+                    manifest = {
+                        "wasm": [{
+                            "data": f.read()
+                        }],
+                    }
```
Comment (bhelx) [inline extism_cli/__init__.py:598]: I could not get the HttpRequest variant to work in the manifest. might want to follow up and look at why. I think this is okay for the time being.

Comment (zshipko) [reply extism_cli/__init__.py:598]:
```suggestion
                    manifest = {
                        "wasm": [{
                            "url": args.wasm
                        }],
                    }
```
^ This worked for me, what error are you getting?

Comment (bhelx) [reply extism_cli/__init__.py:598]: hmm okay, i think the docs are wrong then: https://extism.org/docs/concepts/manifest/
it says there is a top level `req` key. But looking at the code it seems to be flattened into the level above it.

Comment (bhelx) [reply extism_cli/__init__.py]: I'll follow up with doc change if needed

Comment (zshipko) [reply extism_cli/__init__.py]: i see, yeah looks like the docs are wrong. you can reference the generated jsonschema: https://github.com/extism/extism/blob/59101d80a6d80b44a7762d860c523cbb19d07f03/manifest/schema.json#L123

**Review by zshipko — APPROVED** [review]

**Pull request merged**